cmd/attest.go

```diff
@@ -0,0 +1,193 @@
+by --format flag.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		rwc, err := openTpm()
+		if err != nil {
+			return err
+		}
+		defer rwc.Close()
+
+		createFunc := attestationKeys[key][keyAlgo]
+
+		attestationKey, err := createFunc(rwc)
+		if err != nil {
+			return fmt.Errorf("failed to create attestation key: %v", err)
+		}
+		defer attestationKey.Close()
+
+		attestopts := client.AttestOpts{}
+		attestopts.Nonce = nonce
+		if len(teenonce) != 0 {
+			attestopts.TEENonce = teenonce
+			attestopts.TEEDevice, err = client.CreateSevSnpDevice()
+			if err != nil {
+				return fmt.Errorf("collecting TEE attestation report: %v", err)
+			}
+		}
+
+		if ExternalTPM != nil {
+			if elg, ok := ExternalTPM.(client.EventLogGetter); ok {
+				attestopts.TCGEventLog, err = elg.EventLog()
+				if err != nil {
+					return fmt.Errorf("failed to retrieve TCG Event Log: %w", err)
+				}
+			}
+		}
+
+		attestation, err := attestationKey.Attest(attestopts)
+		if err != nil {
+			return fmt.Errorf("failed to collect attestation report: %v", err)
+		}
+
+		if key == "gceAK" {
+			getMetadata()
+			addGceInstanceInfo(attestation)
+		}
+
+		if len(output) != 0 {
+			var out []byte
+			if format == "binarypb" {
+				out, err = proto.Marshal(attestation)
+				if err != nil {
+					return fmt.Errorf("failed to marshal attestation proto: %v", attestation)
+				}
+			} else {
+				out = []byte(prototext.Format(attestation))
+			}
+			if _, err := dataOutput().Write(out); err != nil {
+				return fmt.Errorf("failed to write attestation proto: %v", err)
+			}
+		}
+		return nil
+	},
+}
+
+func getMetadata() error {
+
+	var err error
+	gceProjectID, err = metadata.ProjectID()
+	if err != nil {
+		return err
+	}
+
+	projectNumber, err := metadata.NumericProjectID()
+	if err != nil {
+		return err
+	}
+	gceProjectNumber, err = strconv.ParseUint(projectNumber, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	gceZone, err = metadata.Zone()
+	if err != nil {
+		return err
+	}
+
+	instanceID, err := metadata.InstanceID()
+	if err != nil {
+		return err
+	}
+	gceInstanceID, err = strconv.ParseUint(instanceID, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	gceInstanceName, err = metadata.InstanceName()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+func addGceInstanceInfo(attestation *attest.Attestation) {
+	attestation.InstanceInfo = &attest.GCEInstanceInfo{
+		Zone:          gceZone,
+		ProjectId:     gceProjectID,
+		ProjectNumber: gceProjectNumber,
+		InstanceName:  gceInstanceName,
+		InstanceId:    gceInstanceID,
+	}
+}
+
+func addFormatFlag(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&format, "format", "binarypb", "type of output file where attestation report stored <binarypb|prototext>")
+}
+func addKeyFlag(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVar(&key, "key", "AK", "indicates type of attestation key to use <gceAK|AK>")
+}
+func addNonceflag(cmd *cobra.Command) {
```

Comment:
Since it's shared, this should go in `flags.go`

Reply:
Yes , it is a shared flag and it should be in flags.go .
I'll move it in flags.go.

Reply:
I have moved `--nonce` flags in flags.go.